src/components/Chloropleth.jsx

```diff
@@ -161,39 +162,44 @@ const Chloropleth = (props) => {
   }, getDependencyArray(viewport))
 
   const onViewportChange = newViewport => {
-    clampViewport(newViewport, tiles.config.bounds)
+    clampViewport(newViewport, config.bounds)
     setViewport(newViewport)
   }
 
-  const data = useMemo(() => {
-    if (tiles === null) {
-      return null
+  const features = useMemo(() => {
+    const features = {
+      selected: [],
+      active: [],
+      nulls: [],
+      others: []
     }
 
-    if (index === null) {
-      return tiles
+    if (values === null) {
+      return features
     }
 
-    const features = tiles.features.map(f => {
-      const { area_id } = f.properties
-
-      const values = index ? index[area_id] : null
-      const value = values ? values[date] : undefined
-      return {
-        ...f,
-        properties: {
-          ...f.properties,
-          value,
-          selected: area_id === selected_area
+    for (const feature of geojson.features) {
+      const { area_id } = feature.properties
+      const value = values[area_id]
+      if (value === null) {
+        features.nulls.push(feature)
+        if (area_id === selected_area) features.selected.push(feature)
+      } else if (value !== undefined) {
+        const _feature = {
+          ...feature,
+          properties: {
+            ...feature.properties,
+            value
+          }
         }
+        features.active.push(_feature)
+        if (area_id === selected_area) features.selected.push(feature)
```

Comment:
decided duplicating the selected feature to draw the thick line was the best option for cleaner code and results, don't think it creates a performance issue. The selected feature would flash white if its fill was defined in a separate layer.